### PR TITLE
[chore] Upgrade tap/jest/babel/gulp setup to latest versions

### PR DESCRIPTION
**Pull request by rexxars**
We're having tests fail on travis. After looking into it, it seems we're running a range of different versions of jest. The babel setup didn't work as intended with the latest version of babel, so had to upgrade that. This broke tap, so it had to be updated. I then added node 12 to the test matrix, and it turns out that gulp 3 doesn't work on node 12, so had to upgrade that and rewrite the entire gulpfile.

Long story short, this makes sure that we can run tests on travis again.

### CODE REVIEW

**Comment by skogsmaskin** [comment]
When I ran `test-all` the two first times it failed on the `state-router` and `observable` pacakge. After running the tests inside their folders, they passed, and then `test-all` passed. Not sure what is going on there.

**Comment by thomax** [comment]
Hm. Done a full clean/install/bootstrap on this branch. Running `npm run test-all` and keep getting
```
@sanity/client: Error: Cannot find module './lib/SanityObservableMinimal'
@sanity/client:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
```

**Comment by rexxars** [comment]
@thomax Can you run `npx gulp` and re-run the tests, to see if that changes anything?

**Comment by thomax** [comment]
`npx gulp` did the trick. All lights are green!
